specs/electra/beacon-chain.md

```diff
@@ -1090,6 +1100,20 @@ def process_withdrawals(state: BeaconState, payload: ExecutionPayload) -> None:
 
 #### Execution payload
 
+##### New `compute_execution_requests_hash`
+
+*Note*: Computes commitment to the execution layer triggered requests data.
+The computation algorithm is defined by [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685).
+
+```python
+def compute_execution_requests_hash(execution_requests: ExecutionRequests) -> Hash32:
+    deposit_bytes = DEPOSIT_REQUEST_TYPE + serialize(execution_requests.deposits)
+    withdrawal_bytes = WITHDRAWAL_REQUEST_TYPE + serialize(execution_requests.withdrawals)
+    consolidation_bytes = CONSOLIDATION_REQUEST_TYPE + serialize(execution_requests.consolidations)
+
+    return hash(deposit_bytes + withdrawal_bytes + consolidation_bytes)
```

Comment:
Would it be good to add an assert to this function to ensure the requests are sorted?

Reply:
Doesn't the assert happen here via the engine api? `assert execution_engine.verify_and_notify_new_payload`

Reply:
As of this PR, `verify_and_notify_new_payload` is only given the requests hash, so I'm not sure how it would know if they were in the wrong order. Also, this function in the executable specs is hardcoded to return true. Is `verify_and_notify_new_payload` in `spec_builders` overwritten by this one in the spec?

https://github.com/ethereum/consensus-specs/blob/bda895715390ce06bafced68982102f4614eb223/pysetup/spec_builders/electra.py#L54-L56

Reply:
Yes. That's right. So it's fine how it is. But just to be clear, if in the future we were to add a new request type in the wrong order, the consensus spec tests will not be able to detect it.